product_family/models/product_family.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# (c) 2017 Consultoría Informática Studio73 SL (contacto@studio73.es)
+#          Pablo Fuentes <pablo@studio73.es>
+#
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from openerp import models, fields, api, _
+from openerp.exceptions import ValidationError
+
+
+class ProductFamily(models.Model):
+    _name = 'product.family'
+    _parent_name = "parent_id"
+    _order = 'parent_left'
+    _parent_order = "sequence asc, name"
+    _parent_store = True
+
+    sequence = fields.Integer(string="Sequence")
+    name = fields.Char('Nombre', required=True, translate=True)
+    parent_id = fields.Many2one(comodel_name="product.family",
+                                string="Parent Family", ondelete='cascade')
+    parent_left = fields.Integer('Left Parent', index=True)
+    parent_right = fields.Integer('Right Parent', index=True)
+    type = fields.Selection(selection=[('view', 'View'), ('normal', 'Normal')],
+                            string="Family Type", default="normal")
+
+    @api.constrains('parent_id')
+    def _check_category_recursion(self):
+        if not self._check_recursion():
```

Comment:
where is this function defined ?

Reply:
Is defined at Odoo BaseModel class
https://github.com/odoo/odoo/blob/8.0/openerp/models.py#L4924

Reply:
Thanks ! I did'nt know !